src/iuse.cpp

```diff
@@ -1666,7 +1666,15 @@ int iuse::remove_all_mods( player *p, item *, bool, const tripoint & )
     }
 
     auto loc = g->inv_map_splice( []( const item & e ) {
-        return !e.toolmods().empty();
+        if( e.toolmods().empty() ) {
```

Comment:
Checking this special case in unnecessary. If `toolmods()` returns an empty container, the loop below will essentially skip it and the lambda returns false.